// Copyright 2023 Yi Huang. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package signserver;

// The Inceptio code signing service definition
service CodeSigning {
  // Interface for getting the RSA signature
  rpc GetRsaSignature (RsaSignRequest) returns (RsaSignReply) {}

  // Interface for getting the RSA public key
  rpc GetRsaPublicKey (RsaPubkeyRequest) returns (RsaPubkeyReply) {}
}

// The request message containing the info for a RSA signature
message RsaSignRequest {
  string hash_str = 1;
  int32 key_set = 2;
  int32 key_id = 3;
}

// The response message containing the RSA signature hex string
message RsaSignReply {
  string signature = 1;
}

// The request message containing the info for a RSA public key
message RsaPubkeyRequest {
  int32 key_set = 1;
  int32 key_id = 2;
  int32 key_type = 3;
}

// The response message containing the RSA public key
message RsaPubkeyReply {
  string public_key = 1;
}
